solutions/0000-0099/0036-valid-sudoku-easy.md

```diff
@@ -1,5 +1,168 @@
 ---
-draft: true
+description: >-
+  Author: @vigneshshiv |
+  https://leetcode.com/problems/valid-sudoku/
 ---
 
 # 0036 - Valid Sudoku (Easy)
+
+## Problem Link
+
+https://leetcode.com/problems/valid-sudoku/
+
+## Problem Statement
+
+Determine if a `9 x 9` Sudoku board is valid. Only the filled cells need to be validated **according to the following rules:**
+
+1. Each row must contain the digits `1-9` without repetition.
+2. Each column must contain the digits `1-9` without repetition.
+3. Each of the nine `3 x 3` sub-boxes of the grid must contain the digits `1-9` without repetition.
+
+**Note:**
+* A Sudoku board (partially filled) could be valid but is not necessarily solvable.
+* Only the filled cells need to be validated according to the mentioned rules.
+
+**Example 1:**
+
+```
+5 3 .  . 7 .  . . .  
+6 . .  1 9 5  . . .  
+. 9 8  . . .  . 6 .  
+
+8 . .  . 6 .  . . 3  
+4 . .  8 . 3  . . 1  
+7 . .  . 2 .  . . 6  
+
+. 6 .  . . .  2 8 .  
+. . .  4 1 9  . . 5  
+. . .  . 8 .  . 7 9 
+```
+
+```
+Input: board = 
+[["5","3",".",".","7",".",".",".","."]
+,["6",".",".","1","9","5",".",".","."]
+,[".","9","8",".",".",".",".","6","."]
+,["8",".",".",".","6",".",".",".","3"]
+,["4",".",".","8",".","3",".",".","1"]
+,["7",".",".",".","2",".",".",".","6"]
+,[".","6",".",".",".",".","2","8","."]
+,[".",".",".","4","1","9",".",".","5"]
+,[".",".",".",".","8",".",".","7","9"]]
+Output: true
+```
+
+**Example 2:**
+
+```
+Input: board = 
+[["8","3",".",".","7",".",".",".","."]
+,["6",".",".","1","9","5",".",".","."]
+,[".","9","8",".",".",".",".","6","."]
+,["8",".",".",".","6",".",".",".","3"]
+,["4",".",".","8",".","3",".",".","1"]
+,["7",".",".",".","2",".",".",".","6"]
+,[".","6",".",".",".",".","2","8","."]
+,[".",".",".","4","1","9",".",".","5"]
+,[".",".",".",".","8",".",".","7","9"]]
+Output: false
+Explanation: Same as Example 1, except with the 5 in the top left corner being modified to 8. Since there are two 8's in the top left 3x3 sub-box, it is invalid.
+```
+
+**Constraints:**
+
+* `board.length == 9`
+* `board[i].length == 9`
+* `board[i][j]` is a digit `1-9` or `'.'`.
+
+## Approach 1: HashSet
+
+Every value in the row or column or `3 x 3` block, the value is unique, With the help of this simple logic, add the Numbers in the HashSet, Add `num` value for `m`th row and for `n`th column and the same for `3 x 3` matrix. 
+
+If the same value repeat again for the same row or column or `3 x 3` block, then the given sudoku is not valid, otherwise is valid.
+
+Time Complexity: O(m * n), where m - # of rows, n - # of cols
```

Comment:
are u using 0 for Big O notation?

Reply:
I'm using a Big O notation, anyway i will change with LaTex for easy to read.